cf/__init__.py

```diff
@@ -95,154 +94,141 @@
     "packaging",
     "scipy",
 )
-
 x = ", ".join(_requires)
 _error0 = f"cf v{__version__} requires the modules {x}. "
 
+import importlib.util
+from platform import python_version
+
+_found_esmpy = bool(importlib.util.find_spec("esmpy"))
+
 try:
-    import cfdm
+    import packaging
+    from packaging.version import Version
 except ImportError as error1:
     raise ImportError(_error0 + str(error1))
+else:
+    _minimum_vn = "20.0"
+    if Version(packaging.__version__) < Version(_minimum_vn):
+        raise RuntimeError(
+            f"Bad packaging version: cf requires packaging>={_minimum_vn}. "
+            f"Got {packaging.__version__} at {packaging.__file__}"
+        )
 
-__cf_version__ = cfdm.core.__cf_version__
-
-from packaging.version import Version
-import importlib.util
-import platform
-
-# ESMF renamed its Python module to `esmpy` at ESMF version 8.4.0. Allow
-# either for now for backwards compatibility.
-_found_esmpy = bool(
-    importlib.util.find_spec("esmpy") or importlib.util.find_spec("ESMF")
-)
+try:
+    import cfdm
+except ImportError as error1:
+    raise ImportError(_error0 + str(error1))
+else:
+    # Check the version of cfdm
+    _minimum_vn = "1.12.0.0"
+    _maximum_vn = "1.12.1.0"
+    _cfdm_version = Version(cfdm.__version__)
+    if not Version(_minimum_vn) <= _cfdm_version < Version(_maximum_vn):
```

Comment:
To avoid a mix of `not` negation and the comparison operators:

```suggestion
    if (
        _cfdm_version < Version(_minimum_vn) or
        _cfdm_version > Version(_maximum_vn)
    ):
```